cfdm/docstring/docstring.py

```diff
@@ -220,6 +220,39 @@
                 ``3``/``'DETAIL'``) for increasing verbosity, the more
                 description that is printed to convey information
                 about the operation.""",
+    "{{d: `int`}}": """d: `int`
+                The position in the tie points array of the subsampled
+                dimension being interpolated.""",
+    # d1
+    "{{d1: `int`}}": """d1: `int`
+                The position of subsampled dimension 1 (in the sense
+                of CF Appendix J Figure J.2) in the full tie points
+                array.""",
+    # d2
+    "{{d2: `int`}}": """d2: `int`
+                The position of subsampled dimension 2 (in the sense
+                of CF Appendix J Figure J.2) in the full tie points
+                array.""",
+    # s
+    "{{s: array_like, optional}}": """s: array_like, optional
+                If set to a single number in the range [0, 1] then
+                this value is return for the interpolation coeficient
+                ``s``. By default ``s`` is calculated for each of the
+                interpolation subararea's uncompressed location.""",
+    # s_i
+    "{{s_i: array_like}}": """s_i: array_like
+                A value for the interpolation coeficient ``s`` for the
+                subsampled dimension, at some the location between the
```

Comment:
I am not sure what 'at some...' is trying to say?

Reply:
The interpolation coefficient `s` for a pair of tie points varies strictly monotonically from 0 to 1, with values 0 and 1 being at  the locations of the tie points themselves. `s` has values that correspond to the locations of each uncompressed grid point that lies between the two tie points, and varies linearly in the index space of the uncompressed grid.

Some functions require as an input the value of `s` that corresponds to a particular index location between the two tie points, e.g. `s=0.5` is the interpolation coefficient for "half-way" between the tie points. (I put half-way in quotes because if a value 0.5 won't match up with an uncompressed grid point if there are an even number of coefficient values, e.g. `s=[0, 0.2, 0.4, 0.6, 0.8, 1.0]` has 6 values, none of which are 0.5. However, the relevant equations are designed to not mind about this move to a continuum.)

How about:

```python
    "{{s_i: array_like}}": """s_i: array_like
                A value for the interpolation coefficient ``s`` for the
                subsampled dimension that corresponds to a particular
                uncompressed location between the two tie points.

                *Parameter example:*
                  ``0.5``""",
```

Reply:
I see, thanks for clarifying in detail! Your suggestion seems very clear and concise, so please go ahead with that.

Reply:
Fixed in https://github.com/NCAS-CMS/cfdm/pull/168/commits/fdd65c2f326a25e7812e79c592b79229391ca1c3